### PR TITLE
libwebsockets: add compile options

### PR DESCRIPTION
**Pull request by tsl0922**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Add some common compile options for the libwebsockets package.

### CODE REVIEW

**Comment by ilovezfs** [comment]
Which ones are you going to use?

**Comment by tsl0922** [comment]
`--with-libuv` and `--with-unix-sock`, and maybe `--without-test-apps`.

**Comment by ilovezfs** [comment]
OK let's make all of those default and forget about the options.

**Comment by tsl0922** [comment]
ping @ilovezfs

File: Formula/libwebsockets.rb [hunk 1]
```diff
@@ -13,9 +14,23 @@ class Libwebsockets < Formula
...
+      -DLWS_WITH_LIBUV=ON
+      -DLWS_WITH_LIBEVENT=ON
+      -DLWS_WITH_HTTP2=ON
+      -DLWS_IPV6=ON
+      -DLWS_UNIX_SOCK=ON
+      -DLWS_WITH_PLUGINS=ON
+      -DLWS_WITHOUT_TESTAPPS=ON
+    ]
+
+    system "cmake", ".", *cmake_args
```
Comment (fxcoudert) [inline Formula/libwebsockets.rb]: As a matter of style, just do:

```
  system "cmake", ".", *std_cmake_args,
                  "-DLWS_WITH_LIBEV=ON",
                  …
```

Comment (tsl0922) [reply Formula/libwebsockets.rb]: Updated as #21653, the old fork is deleted so I can't update this PR anymore.

**Pull request closed without merging**